### ISSUE TITLE
Very strange GANed cats

### ISSUE THREAD

**Issue by MNCTTY** [jhayes14/GAN#6]
Hello! 
Thank you for you implementation, it was really cool expierence to run it.
But I've got some strange results.
I've trained the model on dataset with cats, there were 200 pics on 100 epochs with batch size 10. And here what it's generated.
I really don't know how to interpret the results 
![4 step 5batch 100 epochs _ 1](https://user-images.githubusercontent.com/37251686/39298534-221667b0-494f-11e8-99c9-28ae7181c7cf.jpg)
![4 step 5batch 100 epochs](https://user-images.githubusercontent.com/37251686/39298535-22760ec2-494f-11e8-98c6-b94ef1dac02e.jpg)

**Comment by jhayes14** [comment]
Hi, it could be for a hundred reasons tbh. Maybe mode collapse? Do the losses get stuck at some values?

**Comment by MNCTTY** [comment]
Yes, it first strange thing: Generator loss was 1.1920933e-07 almost all time , and Discriminator loss was changing from one batch to the next one.
Second thing, it looks like model doesn't train a lot on batches. Here some text from terminal 
```

Epoch 1 Batch 1856
Training first discriminator..
Training first generator..
Initial batch losses :  Generator loss 1.1920933e-07 Discriminator loss 1.4226825e-07 Total: 2.6147757e-07
Final batch losses (after updates) :  Generator loss 1.1920933e-07 Discriminator loss 1.4226825e-07 Total: 2.6147757e-07

Epoch 1 Batch 1857
Training first discriminator..
Training first generator..
Initial batch losses :  Generator loss 1.1920933e-07 Discriminator loss 4.04652e-05 Total: 4.0584408e-05
Final batch losses (after updates) :  Generator loss 1.1920933e-07 Discriminator loss 4.04652e-05 Total: 4.0584408e-05

Epoch 1 Batch 1858
Training first discriminator..
Training first generator..
Initial batch losses :  Generator loss 1.1920933e-07 Discriminator loss 1.0960467e-07 Total: 2.28814e-07
Final batch losses (after updates) :  Generator loss 1.1920933e-07 Discriminator loss 1.0960467e-07 Total: 2.28814e-07
```

When I run it very first time, I had a lot of string of text on each batch, but from some time - only this.
This text is from the other code run (on whole dataset, 2000 batches with theirs size 2, and 2 epoches), but that time quantity of text lines was the same.

**Comment by jhayes14** [comment]
Your losses are essentially zero, which means it is not learning. Try a larger batch size. After that I would honestly try a more well-tested implementation such as https://github.com/jacobgil/keras-dcgan , since I have optimised this implementation at all.

**Comment by MNCTTY** [comment]
Ok, thank you, I'll try.
By the way, I tried to use batch_size 50, 100 and 200 and 2 epochs. And, it generates something new, but it doesnt look like a cat :)
But I saw how losses really changes
![batch size 200 epochs 2 _____ 0](https://user-images.githubusercontent.com/37251686/39500573-2a350154-4dbe-11e8-98ee-2f59ccf75b5d.jpg)
